### PR TITLE
refactor: es6-fy

### PR DESCRIPTION
**Pull request by curbengh**

### CODE REVIEW

File: lib/deployer.js [hunk 1]
```diff
@@ -48,10 +48,10 @@ module.exports = function(args) {
   }
 
   function git() {
-    var len = arguments.length;
-    var args = new Array(len);
+    const len = arguments.length;
```
Comment (segayuu) [inline lib/deployer.js]: I propose to convert to rest parameters.

File: lib/parse_config.js [hunk 2]
```diff
@@ -27,20 +27,20 @@ function parseRepo(repo) {
...
+    const data = parseRepo(repo);
     data.branch = args.branch || data.branch;
 
     return [data];
   }
 
-  var result = [];
-  var keys = Object.keys(repo);
+  const result = [];
+  const keys = Object.keys(repo);
```
Comment (segayuu) [inline lib/parse_config.js]: I suggest converting to code that uses `Array.prototype.map()`.

**Comment by curbengh** [comment]
spread syntax pending https://github.com/hexojs/hexo-deployer-git/pull/138

File: lib/deployer.js (hunk 1 above)
Comment (curbengh) [reply lib/deployer.js]: I refactored to spread, I'll change it to rest.

Comment (segayuu) [reply lib/deployer.js]: I do not know why you expand arguments object with spread syntax without using rest parameters for git.
I would be glad if you could answer the reason :)

**Comment by segayuu** [comment]
A new lint error can be avoided by adding `"engines": {"node": ">=8"}` in package.json :)
https://docs.npmjs.com/files/package.json#engines

File: lib/deployer.js (hunk 1 above)
Comment (curbengh) [reply lib/deployer.js]: indeed, it should be [rest](https://stackoverflow.com/questions/960866/how-can-i-convert-the-arguments-object-to-an-array-in-javascript).

**Comment by curbengh** [comment]
> A new lint error can be avoided by adding `"engines": {"node": ">=8"}` in package.json :)
> https://docs.npmjs.com/files/package.json#engines

Added by https://github.com/hexojs/hexo-deployer-git/pull/138. Once that is merged, I can rebase this PR, and the test should pass.

**Pull request merged**